<!DOCTYPE html>
<html lang="en">
<head>
	<meta charset="utf-8">
	<meta content="IE=edge,chrome=1" http-equiv="X-UA-Compatible">
	<meta content='width=device-width, initial-scale=1.0' name='viewport' />
	<title>FISHR Project</title>
	<meta property="og:title" content="KellerLab + FISHR Project" />
    <meta property="og:type" content="website" />
    <meta property="og:url" content="http://kellerfishrproject.com" />
    <meta property="og:description" content="FISHR Project is a tool for geneticists for rapidly locating IBD shared haplotypes." />
    <link rel="stylesheet" type="text/css" href="public/css/site_normalize.css">
	<link rel="stylesheet" type="text/css" href="public/css/site_style.css">
	<link href='http://fonts.googleapis.com/css?family=Montserrat:400,700' rel='stylesheet' type='text/css'>
	<script type="text/javascript" src="http://use.typekit.net/cgj8miu.js"></script>
	<script type="text/javascript">try{Typekit.load();}catch(e){}</script>
	<!--[if lt IE 9]>
            <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
            <link href="/public/css/ie.min.css" rel="stylesheet" type="text/css" media="all"/>
        <![endif]-->
</head>
<body>
 <!--[if lt IE 7]>
        <p class="chromeframe">You are using an <strong>outdated</strong> browser. Please <a href="http://browsehappy.com/">upgrade your browser</a> or <a href="http://www.google.com/chromeframe/?redirect=true">activate Google Chrome Frame</a> to improve your experience.</p>
    <![endif]-->
	<section id="intro">
		<header id="stick">
			<div class="container">
				<div id="logo">
					<a href="home.html"><h1>KellerLab + FISHR Project</h1></a>
				</div>
				<nav id="nav">
					<ul>
						<li><a href="#documentation">Overview</a></li>
						<li><a href="#germline">Germline</a></li>
						<li><a href="#errorfinder">Fishr</a></li>
						<li><a href="#errorfinderlowram">Low-Ram</a></li>
						<li class="last"><a href="#glparameterfinder">GL Parameter Finder</a></li>
					</ul>
				</nav>
			</div>
		</header>
		<div class="tag">
			<h2>THE FISHR PROJECT</h2>
			<h3>Find IBD Shared Haplotypes Rapidly</h3>
		</div>
	</section> <!-- End of intro section-->
	<section id="documentation">
		<div class="container">
			<div class="content">
				<h2>User Documentation</h2>
				<p>This page contains information for how to use and understand FISHR from the perspective of a user. For information pertaining to software developers, see the <a href="dev_documentation.html">developer documentation</a> page<p>
				<p>Table of contents:</p>
				<ul>
					<li><a href="#overview">FISHR Suite Overview</a></li>
					<li><a href="#germline">Germline</a></li>
					<li><a href="#errorfinder">Fishr (Error Finder)</a></li>
					<li><a href="#errorfinderlowram">Fishr Low-RAM</a></li>
					<li><a href="#glparameterfinder">GL Parameter Finder</a></li>
				</ul> 
			</div>
		</div>
	</section>
	<section id="overview">
		<div class="container">
			<div class="content">
				<h2>FISHR Suite Overview</h2>
				<p>FISHR is a package of software programs that performs statistical analaysis on genetic data to find shared haplotype segements that are identical by descent (IBD).</p>
				<p>It consits of three major programs (Germline, GL parameter finder, Fishr). Note that sometimes, the programs Fishr and ErrorFinder are used interchangeably.</p>
			</div>
		</div>
	</section>
	<section id="germline">
		<div class="container">
			<div class="content">
				<h2>Germline</h2>
				<p>The first step in using FISHR is to use another program called GERMLINE, details of which can be found <a href="http://www1.cs.columbia.edu/~gusev/germline/">here</a>.
GERMLINE is a program that was developed at the University of Columbia. It discovers long shared IBD segments between pairs of individuals in a large population. GERMLINE requires genotype or haplotype data as its inputs, and can execute over phased or un-phased data. Depending on which parameters you give it, GERMLINE will produce a different set of output files. For the purposes of FISHR, we can use GERMLINE to produce binary output files, which results in the production of three files, *.bsid, *.bmid, *.bmatch. These files, along with a .ped file, will be the inputs into FISHR. For more information on these file types, please see the GERMLINE documentation linked above. We use a slightly customized version of GERMLINE that supports a single command line input, rather than the default interactive input. There are a couple of things to take note of in our customized version Germline:</p>
				<ul>
					<li><span class="list-bold">Using '-haploid -h_extend -w_extend ': </span></br>For each strand in a person it will make a separate person. So between two persons there can exist 4 matches in a particular region. This means that phase/SNP errors will break up SHs. For same-person SHs (ROHs), it will output the full genome length twice as well as each ROH twice.</li></br>
					<li><span class="list-bold">Using '-h_extend -w_extend': </span></br>Between two persons, if any one of the two strands from person 1 and any one of the two strands from person 2 matches exactly in the initial seeds, then that will become a SH. It then extends forwards or backwards using the err_hom argument for mismatches, until there are too many mismatches. It allows for phase errors, but only if they occur exactly at the breakpoints of the bits. ROHs are output only once per ROH, and full genome length is not included as an ROH.</li></br>
					<li><span class="list-bold">Using only '-w_extend':  </span></br>First, it will check for exact matches in the initial seeds, similar to using '-h_extend -w_extend'. Once a seed matches, it extends in the forward and backward direction until the first mutually opposite homozygote match is found. ROHs are output only once per ROH, and full genome length is not included as an ROH. </li></br>
				</ul>
<p> A sample of using GERMLINE to prepare data for FISHR would look like:  </p>
				<pre class="codeblock-wrapper">
					<code class="documentation-code-block">
germline8 -pedfile [pedfile] -mapfile [mapfile] -outfile [output format of your choosing] -bin_out -err_hom [arguments] -err_het [arguments] -reduced -bits [some int value] -min_m [minimum length in cM] [method] -homoz
					</code>
				</pre>
				<p>Now, we are finally ready to run FISHR. 
As an aside, if you ever need to make changes to the current version of Germline (say, to add further customization), the way to compile Germline from the command line is:<p>
				<pre class="codeblock-wrapper">
					<code class="documentation-code-block">
g++ -O2 -o [New GERMLINE Version] -I include GERMLINE_0001.cpp GERMLINE.cpp Share.cpp Chromosome.cpp ChromosomePair.cpp HMIndividualsExtractor.cpp MarkerSet.cpp Individual.cpp Individuals.cpp InputManager.cpp MatchFactory.cpp MatchesBuilder.cpp NucleotideMap.cpp PEDIndividualsExtractor.cpp Match.cpp PolymorphicIndividualsExtractor.cpp SNP.cpp SNPPositionMap.cpp SNPs.cpp
					</code>
				</pre>
			</div>
		</div>
	</section>
	<section id="errorfinder">
		<div class="container">
			<div class="content">
				<h2>Fishr (Error Finder)</h2>
				<p>Once the data has been acquired and ran through GERMLINE, we are ready to begin processing it with one of the programs from the FISHR suite. FISHR currently supports three main programs:</p>
				<ul>
					<li>Fishr (Error Finder)</li>
					<li>Low-Ram</li>
					<li>GL_parameter_finder</li>
				</ul>
				<p>Each of these programs are described in more detail in their respective sections of this document. Fishr is responsible for performing the main statistical calculations. Low-Ram is a version of Fishr that is optimized for running in low-memory environments. GL_Parameter_Finder is used for determining certain empirical inputs to the Fishr/Low-Ram programs.</p>
				<h2>Fishr instructions</h2>
				<p>ErrorFinderMain (version 23.5 as of this writing) performs a lot of computations. It is run from the command line, and takes a set of files produced by GERMLINE as its input. The files are:</p>
				<ul>
					<li>*.bmid</li>
					<li>*.bsid</li>
					<li>*.bmatch</li>
					<li>*.ped</li>
				</ul>
				<p>In addition to these files, Fishr also takes several command line arguments.
 Let’s start by looking at how it can be run from the command line:</p>
 <pre class="codeblock-wrapper">
					<code class="documentation-code-block">
ErrorFinder23.5 -bmatch FISHR.errhigh.bit96.ErrHom2.ErrHet0.Option1.MinCM0.5.gp1.N1k.bmatch -bsid FISHR.errhigh.bit96.ErrHom2.ErrHet0.Option1.MinCM0.5.gp1.N1k.bsid -bmid FISHR.errhigh.bit96.ErrHom2.ErrHet0.Option1.MinCM0.5.gp1.N1k.bmid -reduced 100 2.0 -ped-file /work/KellerLab/mmkeller/Find.Optimal.Pihats/Simulated.Data/Final.Phased.Data2/Beagle.Phased.Group1.1k.ped -window 50 -ma-threshold 0.2 -gap 2 -pct-err-threshold 0.8 -trueCM 6 -trueSNP 500 -count.gap.errors TRUE -PIE.dist.length 3 -output.type finalErrorsOutput -log.file s8v2t1 | gzip > s8v2t1.gz
					</code>
				</pre>
			</br>
				Click <span id="toggle-arguments">here</span> to display a complete list of the command line arguments that Fishr accepts. Default parameters for certain values are listed in ():
				<div class="hidden-div-fishr-args"> 
					<ul>
					<li><strong>-bmatch:</strong> This is the path to the bmatch file to be used by ErrorFinder.</li>
					<li><strong>-bsid:</strong> This is the path to the bsid file.</li>
					<li><strong>-bmid:</strong> This is the path to the bmid file.</li>
					<li><strong>-ped-file:</strong> This is the path to the ped file.</li>
					<li><strong>-reduced (snp)(cM):</strong> The reduced option takes two values, the first is the minimum acceptable SNP length, and the second is the minimum acceptable cM length. Every SH will be checked against these values. If a SH has a SNP length OR a cM length below the reduced value, it will be dropped and will not undergo any further processing by ErrorFinder.</li>
					<li><strong>-window:</strong> This specifies the length in SNPs of the sliding window that is used in the moving averages calculations. We will describe the moving averages algorithm, and why we would want to use it, in more detail in a later section. Generally speaking, a value of 50 snps is used by default.</li>
					<li><strong>-ma-threshold:</strong> This is also used in moving average calculations. Basically, after processing all TrulyIBD data in the data set, a moving averages value is found for each of those TrulyIBD segments, and is stored in a sorted array. This value (between 0.0 and 1.0) is an index into that array. It is saying, “take the Xth percentile of the trulyIBD moving averages array”.For example, if we have 10 truly ibd ma values sorted as [0.1,0.1,0.2,0.2,0.2,0.5,0.6,0.7,0.8,0.9] then a -ma-threshoold value of 70 will give the 70th percentile value, in this case, 0.6.</li>
					<li><strong>-empirical-ma-threshold:</strong> Supplying this parameter bypasses the -ma-threshold calculations entirely, and allows you to specify the threshold you wish to use for moving averages without calculating it via truly ibd segments. It should be noted that the use of -ma-threshold and -empirical-ma-threshold are <strong>mutually exclusive</strong>.</li>
					<li><strong>-gap:</strong> SHs from GERMLINE can be broken up by opposite homozygote SNPs when the underlying segment is IBD due to SNP call errors or to deletions. FISHR looks for same-person SHs that have a gap of K SNPs between them and puts them back together under the assumption that two long SHs that are separated by K or fewer SNPs are likely to be IBD.</li>
					<li><strong>-trueCM M (6):</strong> M is the length in cM for a SH to be deemed truly IBD (for finding pct-err-threshold; see above). If few SHs are expected to exist in the dataset of this cM length (due to a small sample size or high error SNP data), then M may need to be lt 6. Check the line “No of matches assumed to be IBD for deriving -pct-err-threshold and (if applicable) holdout-threshold” in the log file; if this is fewer than 100, then consider lowering M. However, if M gets too small (e.g., lt 5), then non-IBD SHs begin to be counted, increasing the threshold and resulting in more false positive SH calls</li>
					<li><strong>-trueSNP N (600):</strong>N is the length in SNPs for a SH to be deemed truly IBD (for finding pct-err-threshold; see above). If few SHs are expected to exist in the dataset of this SNP length (due to a small sample size or high error SNP data), then N may need to be lt 600. Check the line “No of matches assumed to be IBD for deriving -pct-err-threshold and (if applicable) holdout-threshold” in the log file; if this is fewer than 100, then consider lowering N. However, if N gets too small (e.g., lt 400), then non-IBD SHs begin to be counted, increasing the threshold and resulting in more false positive SH calls.</li>
					<li><strong>-log.file R:</strong>Name of the log file, which outputs parameters, timing, and other relevant information.</li>
					<li><strong>-output.type S:</strong>Controls what type of output is produced. For most situations, this should be “full”; other options are useful for understanding which SH were dropped and why. [In brackets are the number of SHs assuming 10K SHs following GERMLINE, 8K following consolidation, and 2K that were not dropped. Further, say 500 rows were dropped following holdout step]. S can be any of the following:
						<ul>
							<li><strong>finalOutput [default] (aka “Full”)</strong> - Outputs the final called SHs. The columns of this file are pers1, pers2, bp.distance.start, bp.distance.end, no.of.snps.in.match, and cm.distance [2k rows from above; or 1.5k if holdout was used] </li>
							<li><strong>FullPlusDropped</strong> - Works exactly like full but also includes the SH's that were dropped and the reason for the drop. [8k rows]</li>
						</ul>
					</li>
					<li><strong></strong></li>
					<li><strong></strong></li>
					<li><strong></strong></li>
					</ul>
				</div>
			</div>
		</div>
	</section>
	<section id="errorfinderlowram">
		<div class="container">
			<div class="content">
				<h2>Fishr Low-RAM</h2>
				<p>Fishr Low-Ram is almost identical to Fishr, with the exception that it has been optimized for low-RAM environments.
					Specifically, this version will stream data from Germline, as opposed to stroring it in internal data structures, which leads to a substantial savings in RAM. One consequence of this is that data that is outputted is unsorted, so some post-processing will have to be done in a language like Bash or R to sort the results. Note that unless you really need to save RAM, you should use the main version of Fishr.</p>
			</div>
		</div>
	</section>
	<section id="glparameterfinder">
		<div class="container">
			<div class="content">
				<h2>GL Parameter Finder</h2>
				<p>The GL_parameter_finder program is used to find values for the -empirical-ma-threshold and -empirical-error-threshold values in Fishr.</p>
			</div>
		</div>
	</section>
	<footer>
	    <div class="container">
	        <div class="content">
	            <p>© 2010‑2014 KellerLab. All rights reserved.</p>
	        </div>
	    </div>
	</footer>
<script src="http://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
<script>window.jQuery || document.write('<script src="public/js/jquery-1.11.1.min.js"><\/script>')</script>
<script src="public/js/jquery.scrollTo.min.js"></script>
<script src="public/js/jquery.localscroll.min.js"></script>
<script src="public/js/jquery.sticky.min.js"></script>
<script src="public/js/jquery.backstretch.min.js"></script>
<script src="public/js/responsive-nav.min.js"></script>
<script src="public/js/picturefill.min.js"></script>
<script>
	$.localScroll({
	offset: $('#stick').outerHeight() * - 0.7
	});
$('#stick').sticky();
	var navigation = responsiveNav('#nav', {
		animate: true	
	});
$('#intro').backstretch('public/img/cover_web.jpg');
</script>
<script>
$('#toggle-arguments').click(function(){
	$('.hidden-div-fishr-args').toggle();
})
</script>
</body>
</html>
