<!DOCTYPE html>
<html lang="en">
<head>
	<meta charset="utf-8">
	<meta content="IE=edge,chrome=1" http-equiv="X-UA-Compatible">
	<meta content='width=device-width, initial-scale=1.0' name='viewport' />
	<title>FISHR Project</title>
	<meta property="og:title" content="KellerLab + FISHR Project" />
    <meta property="og:type" content="website" />
    <meta property="og:url" content="http://kellerfishrproject.com" />
    <meta property="og:description" content="FISHR Project is a tool for geneticists for rapidly locating IBD shared haplotypes." />
    <link rel="stylesheet" type="text/css" href="public/css/site_normalize.css">
	<link rel="stylesheet" type="text/css" href="public/css/site_style.css">
	<link href='http://fonts.googleapis.com/css?family=Montserrat:400,700' rel='stylesheet' type='text/css'>
	<script type="text/javascript" src="http://use.typekit.net/cgj8miu.js"></script>
	<script type="text/javascript">try{Typekit.load();}catch(e){}</script>
	<!--[if lt IE 9]>
            <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
            <link href="/public/css/ie.min.css" rel="stylesheet" type="text/css" media="all"/>
        <![endif]-->
</head>
<body>
 <!--[if lt IE 7]>
        <p class="chromeframe">You are using an <strong>outdated</strong> browser. Please <a href="http://browsehappy.com/">upgrade your browser</a> or <a href="http://www.google.com/chromeframe/?redirect=true">activate Google Chrome Frame</a> to improve your experience.</p>
    <![endif]-->
	<section id="intro">
		<header id="stick">
			<div class="container">
				<div id="logo">
					<a href="home.html"><h1>KellerLab + FISHR Project</h1></a>
				</div>
				<nav id="nav">
					<ul>
						<li><a href="#about">About</a></li>
						<li><a href="#software">Software</a></li>
						<li><a href="#documentation">Documentation</a></li>
						<li><a href="#downloads">Downloads</a></li>
						<li class="last"><a href="#contacts">Contact</a></li>
					</ul>
				</nav>
			</div>
		</header>
		<div class="tag">
			<h2>THE FISHR PROJECT</h2>
			<h3>Find IBD Shared Haplotypes Rapidly</h3>
		</div>
	</section> <!-- End of intro section-->
	<section id="documentation">
		<div class="container">
			<div class="content">
				<h2>User Documentation</h2>
				<p>This page contains information for how to use and understand FISHR from the perspective of a user. For information pertaining to software developers, see the <a href="dev_documentation.html">developer documentation</a> page<p>
				<p>Table of contents:</p>
				<ul>
					<li><a href="#overview">FISHR Suite Overview</a></li>
					<li><a href="#germline">Germline</a></li>
					<li><a href="#errorfinder">Fishr (Error Finder)</a></li>
					<li><a href="#errorfinderlowram">Fishr Low-RAM</a></li>
					<li><a href="#glparameterfinder">GL Parameter Finder</a></li>
				</ul> 
			</div>
		</div>
	</section>
	<section id="overview">
		<div class="container">
			<div class="content">
				<h2>FISHR Suite Overview</h2>
				<p>FISHR is a package of software programs that performs statistical analaysis on genetic data to find shared haplotype segements that are identical by descent (IBD).</p>
				<p>It consits of three major programs (Germline, GL parameter finder, Fishr). Note that sometimes, the programs Fishr and ErrorFinder are used interchangeably.</p>
			</div>
		</div>
	</section>
	<section id="germline">
		<div class="container">
			<div class="content">
				<h2>Germline</h2>
				<p>The first step in using FISHR is to use another program called GERMLINE, details of which can be found <a href="http://www1.cs.columbia.edu/~gusev/germline/">here</a>.
GERMLINE is a program that was developed at the University of Columbia. It discovers long shared IBD segments between pairs of individuals in a large population. GERMLINE requires genotype or haplotype data as its inputs, and can execute over phased or un-phased data. Depending on which parameters you give it, GERMLINE will produce a different set of output files. For the purposes of FISHR, we can use GERMLINE to produce binary output files, which results in the production of three files, *.bsid, *.bmid, *.bmatch. These files, along with a .ped file, will be the inputs into FISHR. For more information on these file types, please see the GERMLINE documentation linked above. We use a slightly customized version of GERMLINE that supports a single command line input, rather than the default interactive input. There are a couple of things to take note of in our customized version Germline:</p>
				<ul>
					<li><span class="list-bold">Using '-haploid -h_extend -w_extend ': </span></br>For each strand in a person it will make a separate person. So between two persons there can exist 4 matches in a particular region. This means that phase/SNP errors will break up SHs. For same-person SHs (ROHs), it will output the full genome length twice as well as each ROH twice.</li></br>
					<li><span class="list-bold">Using '-h_extend -w_extend': </span></br>Between two persons, if any one of the two strands from person 1 and any one of the two strands from person 2 matches exactly in the initial seeds, then that will become a SH. It then extends forwards or backwards using the err_hom argument for mismatches, until there are too many mismatches. It allows for phase errors, but only if they occur exactly at the breakpoints of the bits. ROHs are output only once per ROH, and full genome length is not included as an ROH.</li></br>
					<li><span class="list-bold">Using only '-w_extend':  </span></br>First, it will check for exact matches in the initial seeds, similar to using '-h_extend -w_extend'. Once a seed matches, it extends in the forward and backward direction until the first mutually opposite homozygote match is found. ROHs are output only once per ROH, and full genome length is not included as an ROH. </li></br>
				</ul>
<p> A sample of using GERMLINE to prepare data for FISHR would look like:  </p>
				<pre class="codeblock-wrapper">
					<code class="documentation-code-block">
germline8 -pedfile [pedfile] -mapfile [mapfile] -outfile [output format of your choosing] -bin_out -err_hom [arguments] -err_het [arguments] -reduced -bits [some int value] -min_m [minimum length in cM] [method] -homoz
					</code>
				</pre>
				<p>Now, we are finally ready to run FISHR. 
As an aside, if you ever need to make changes to the current version of Germline (say, to add further customization), the way to compile Germline from the command line is:<p>
				<pre class="codeblock-wrapper">
					<code class="documentation-code-block">
g++ -O2 -o [New GERMLINE Version] -I include GERMLINE_0001.cpp GERMLINE.cpp Share.cpp Chromosome.cpp ChromosomePair.cpp HMIndividualsExtractor.cpp MarkerSet.cpp Individual.cpp Individuals.cpp InputManager.cpp MatchFactory.cpp MatchesBuilder.cpp NucleotideMap.cpp PEDIndividualsExtractor.cpp Match.cpp PolymorphicIndividualsExtractor.cpp SNP.cpp SNPPositionMap.cpp SNPs.cpp
					</code>
				</pre>
			</div>
		</div>
	</section>
	<section id="errorfinder">
		<div class="container">
			<div class="content">
				<h2>Fishr (Error Finder)</h2>
				<p>Once the data has been acquired and ran through GERMLINE, we are ready to begin processing it with one of the programs from the FISHR suite. FISHR currently supports three main programs:</p>
				<ul>
					<li>Fishr (Error Finder)</li>
					<li>Low-Ram</li>
					<li>GL_parameter_finder</li>
				</ul>
				<p>Each of these programs are described in more detail in their respective sections of this document. Fishr is responsible for performing the main statistical calculations. Low-Ram is a version of Fishr that is optimized for running in low-memory environments. GL_Parameter_Finder is used for determining certain empirical inputs to the Fishr/Low-Ram programs.</p>
			</div>
		</div>
	</section>
	<section id="errorfinderlowram">
		<div class="container">
			<div class="content">
				<h2>Fishr Low-RAM</h2>
				<p>The most up to date version of FISHR can be downloaded here. Older versions can be found on GitHub.</p>
			</div>
		</div>
	</section>
	<section id="glparameterfinder">
		<div class="container">
			<div class="content">
				<h2>GL Parameter Finder</h2>
				<p>The most up to date version of FISHR can be downloaded here. Older versions can be found on GitHub.</p>
			</div>
		</div>
	</section>
	<footer>
	    <div class="container">
	        <div class="content">
	            <p>© 2010‑2014 KellerLab. All rights reserved.</p>
	        </div>
	    </div>
	</footer>
<script src="http://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
<script>window.jQuery || document.write('<script src="public/js/jquery-1.11.1.min.js"><\/script>')</script>
<script src="public/js/jquery.scrollTo.min.js"></script>
<script src="public/js/jquery.localscroll.min.js"></script>
<script src="public/js/jquery.sticky.min.js"></script>
<script src="public/js/jquery.backstretch.min.js"></script>
<script src="public/js/responsive-nav.min.js"></script>
<script src="public/js/picturefill.min.js"></script>
<script>
	$.localScroll({
	offset: $('#stick').outerHeight() * - 0.7
	});
$('#stick').sticky();
	var navigation = responsiveNav('#nav', {
		animate: true	
	});
$('#intro').backstretch('public/img/cover.jpg');
</script>
</body>
</html>
